docs/adr/0005-api-for-spaces.md

```diff
@@ -0,0 +1,107 @@
+
+## Context and Problem Statement
+
+As one of the building blocks for Spaces in oCIS we plan to add an API that returns information about available spaces.
+
+> Note: The term "spaces" is used here in the context of "a space where files can be saved", similar to a directory. It is not to be confused with space in the sense of file space for example.
+
+The purpose of this new API is to give clients a very simple way to query the dynamic list of spaces, that the user has access to. Clients can provide a way better user experience with that.
+
+This API would even allow to provide (WebDAV-) endpoints depending on the kind and version of the client asking for it.
+
+This ADR discusses how this API could be designed.
+
+## Decision Drivers
+
+- Make it easy to work with a dynamic list of spaces of a user for the clients.
+- No longer the need to make assumptions about WebDAV- and other routes in clients.
+- More meta data available about spaces for a better user experience.
+- Part of the bigger spaces plan.
+- Important to consider in client migration scenarios, ie. in CERN.
+
+## Considered Options
+
+1. [Microsoft Graph API](https://developer.microsoft.com/en-us/graph) inspired API that provides the requested information.
+
+## Decision Outcome
+
+This the DRAFT for the API.
+
+### API to Get Info about Spaces
+
+ownCloud servers provide an API to query for available spaces of an user.
+
+Most important, the API returns the WebDAV endpoint for each space. With that, clients do not have to make assumptions about WebDAV routes any more.
+
+See [Drive item in Microsoft Graph API](https://docs.microsoft.com/de-de/graph/api/resources/onedrive?view=graph-rest-1.0)
+
+### Get "Home folder"
+
+Retrieve information about the home space of a user. Note: This is only a subset of the spaces the user might have access to.
+
+API Call: `/me/drive`: Returns the information of the users home folder.
+
+### Get All Spaces of a User
+
+Retrieve a list of available spaces of a user. This includes all spaces the user has access to at that moment, also the home space.
+
+API Call: `/me/drives`: Returns a list of spaces.
+
+### Common Reply
+
+The reply to both calls is either one or a list of [Drive representation objects](https://docs.microsoft.com/de-de/graph/api/resources/drive?view=graph-rest-1.0):
+
+```
+{
+  "id": "string",
+  "createdDateTime": "string (timestamp)",
+  "description": "string",
+  "driveType": "personal | business | documentLibrary",
+  "eTag": "string",
+  "lastModifiedDateTime": "string (timestamp)",
+  "name": "string",
+  "owner": { "@odata.type": "microsoft.graph.identitySet" },
+  "webUrl": "url"
+}
+
+```
+
+The meaning of the object are in ownClouds context:
+
+1. **id** - a unique ID identifying the space
+2. **driveType** - describing the type of the space.
+3. **eTag** - the current ETag of the space
+4. **owner** - an owner object to whom the space belong
```

Comment:
> an owner object to whom the space belong

What about multiple owners?

Reply:
good catch seems that this is limited to one user https://docs.microsoft.com/en-us/graph/api/resources/identityset?view=graph-rest-1.0

Reply:
Yes, this `microsoft.graph.identitySet` seems to only support one user. Strange...

Anyway, either we invent a  list of `coowner` entries, or we allow multiple owners, which I would not suggest because that would break compatibility.

Reply:
@butonic you said we could use groups here, but according to spec only users, devices, application

Reply:
For the CERN use case, we would need to support groups here

Reply:
Hm, the owner is not to be confused with who has access to the space. That has to be defined differently, and I'd say that is out of scope of *this* API. 

For project spaces, I'd say that the owner is usually the "admin god", with some coowners who can apply changes to the access rights. The idea is that spaces should not stick to a person even if he leaves...

Does that make sense?